tests/unit/tls_test.cc

```diff
@@ -1039,8 +1039,12 @@ SEASTAR_THREAD_TEST_CASE(test_reload_certificates) {
             }
 
             try {
-                f2.get();
-                BOOST_FAIL("should not reach");
+                auto res = f2.get();
+                // If the server completes sending data to the client
+                // during the handshake before the client has fully
+                // closed its connection, then the get() call will
+                // succeed by return an empty buffer indicating EOF
+                BOOST_REQUIRE(res.size() == 0);
```

Comment:
Seems like a good change, but not related to the rest of the PR

Reply:
This test was being flaky when using OpenSSL.  It would pass reliably in release but not in debug.

Reply:
nit, might want to use `BOOST_REQUIRE_EQUAL()` for better postmortem debugging experience. as Boost.Test prints out the lhs and rhs of the comparison f the check fails.